### PR TITLE
Fixes #8763: When the conditions contain spaces, there are kept and the generated file is invalid

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://www.rudder-project.org/redmine/issues/8763

### CODE REVIEW

File: builder/index.html [hunk 1]
```diff
@@ -257,7 +258,7 @@ <h4 class="panel-title">
                         <span class="glyphicon glyphicon-plus-sign" style="margin-right:5px"></span>
                         Add methods
                     </div>
-                    <button ng-disabled="editForm.$pending || editForm.$invalid || CForm.form.$invalid || checkSelectedTechnique()" class="btn btn-default pull-right" ng-click="saveTechnique()">Save</button>
+                    <button ng-disabled="editForm.$pending || editForm.$invalid || CForm.form.$invalid || checkSelectedTechnique() || selectedMethod.advanced_class.indexOf(' ')>0 || CForm.versionForm.$invalid" class="btn btn-default pull-right" ng-click="saveTechnique()">Save</button>
```
Comment (VinceMacBuche) [inline builder/index.html]: YOu should not add the check on wetherit contains a space or not, but be based on the fact that Cform.form is $invalid (check advanced class input with a patter)

File: builder/index.html [hunk 2]
```diff
@@ -296,12 +297,21 @@ <h3 >{{getMethodName(selectedMethod)}}
...
+                                </div>
                               </form>
                             </div>
                             <form name="CForm.form">
-                                <div class="form-group condition-form">
-                                  <label for="advanced">Other CFEngine classes:</label>
-                                  <input  type="text" name="cfClasses" id="advanced"  class="form-control" ng-change="updateClassContext()" ng-model="selectedMethod.advanced_class" placeholder="">
+                                <div class="form-group condition-form" ng-class="{'has-error':selectedMethod.advanced_class.indexOf(' ')>0}">
+                                  <label class="control-label" for="advanced">Other CFEngine classes:</label>
+                                  <input type="text" class="form-control" name="cfClasses" id="advanced" aria-describedby="helpBlock" ng-change="updateClassContext()" ng-model="selectedMethod.advanced_class" >
```
Comment (VinceMacBuche) [inline builder/index.html]: You should add a ng-pattern here to check the input ( value should be ng-pattern="/^\S+$/" ) and then you can use angular form validation

File: builder/index.html [hunk 3]
```diff
@@ -296,12 +297,21 @@ <h3 >{{getMethodName(selectedMethod)}}
...
+                                <div ng-messages="CForm.versionForm.versionMin.$error" role="alert">
+                                  <div ng-message="pattern" class="text-danger">Invalid minor version's number</div>
+                                </div>
                               </form>
                             </div>
                             <form name="CForm.form">
-                                <div class="form-group condition-form">
-                                  <label for="advanced">Other CFEngine classes:</label>
-                                  <input  type="text" name="cfClasses" id="advanced"  class="form-control" ng-change="updateClassContext()" ng-model="selectedMethod.advanced_class" placeholder="">
+                                <div class="form-group condition-form" ng-class="{'has-error':selectedMethod.advanced_class.indexOf(' ')>0}">
```
Comment (VinceMacBuche) [inline builder/index.html]: With angular validation you don't need to have the ng-class

File: builder/index.html [hunk 4]
```diff
@@ -296,12 +297,21 @@ <h3 >{{getMethodName(selectedMethod)}}
...
+                                  <label class="control-label" for="advanced">Other CFEngine classes:</label>
+                                  <input type="text" class="form-control" name="cfClasses" id="advanced" aria-describedby="helpBlock" ng-change="updateClassContext()" ng-model="selectedMethod.advanced_class" >
+                                  <span id="helpBlock" class="help-block" ng-show="selectedMethod.advanced_class.indexOf(' ')>0">
```
Comment (VinceMacBuche) [inline builder/index.html]: And you use a ng-message here

**Comment by RaphaelGauthier** [comment]
PR updated

**Pull request merged**